pkg/common/upgrade/releases.go

```diff
@@ -97,13 +85,70 @@ func LatestRelease(releaseStream string, useReleaseControllerForInt bool) (name,
 			continue
 		}
 
-		if latestVersion == nil || currentVersion.GreaterThan(latestVersion) {
-			latestVersion = currentVersion
-			latestCincinnatiRelease = release
+		s.Cache[channel] = append(s.Cache[channel], currentVersion)
+	}
+
+	return nil
+}
+
+var cache *smallCincinnatiCache = &smallCincinnatiCache{
+	Cache: map[string][]*semver.Version{},
+	mutex: sync.Mutex{},
+}
+
+// IsVersionInCincinnati returns true if the version can be found in Cincinnati
+func IsVersionInCincinnati(version *semver.Version) (bool, error) {
+	channel := VersionToChannel(version)
+	cincinnatiVersions, err := cache.Get(channel)
+
+	if err != nil {
+		return false, fmt.Errorf("error loading Cincinnati data: %v", err)
+	}
+
+	for _, cincinnatiVersion := range cincinnatiVersions {
```

Comment:
Instead of a for loop here, we should make cincinatiVersions a map[string]bool and just check to see if the key exists.